src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -795,14 +842,348 @@ def __try_add_error(error_list, detail):
+
+        while status_file_size_in_bytes > self.__internal_file_capacity:
+            # Start truncation process
+            packages_retained_in_assessment, assessment_packages_removed_after_truncation, packages_retained_in_installation, installation_packages_removed_after_truncation = \
+                self.__apply_truncation_process(self.__assessment_packages_copy, self.__installation_packages_copy, size_of_max_packages_allowed_in_status, low_pri_index)
+            if len(assessment_packages_removed_after_truncation) > 0:
+                # Update current assessment # of removed packages
+                self.__assessment_packages_removed = assessment_packages_removed_after_truncation
+                # Recompose truncated status file payload (assessment)
+                truncated_status_file = self.__recompose_truncated_status_file(truncated_status_file, packages_retained_in_assessment, self.__assessment_total_error_count,
+                    self.__assessment_substatus_msg_copy, Constants.PATCH_ASSESSMENT_SUMMARY, assessment_substatus_index)
+
+            if len(installation_packages_removed_after_truncation) > 0:
+                # Update current installation # of removed packages
+                self.__installation_packages_removed = installation_packages_removed_after_truncation
+                # Recompose truncated status file payload (installation)
+                truncated_status_file = self.__recompose_truncated_status_file(truncated_status_file, packages_retained_in_installation, self.__installation_total_error_count,
+                    self.__installation_substatus_msg_copy, Constants.PATCH_INSTALLATION_SUMMARY, installation_substatus_index)
+            status_file_size_in_bytes, status_file_agent_size_diff = self.__get_new_size_in_bytes_after_truncation(truncated_status_file)
+            size_of_max_packages_allowed_in_status -= status_file_agent_size_diff   # reduce the max packages byte size by tombstone, new error, and escape chars byte size
+
+        self.composite_logger.log_debug("End package list truncation")
+        return truncated_status_file
+
+    def __get_new_size_in_bytes_after_truncation(self, truncated_status_file):
+        """ Get new size in bytes for status_file,and difference of status file - 126kb """
+        status_file_size_in_bytes = self.__calc_status_size_on_disk(truncated_status_file)
+        status_file_agent_size_diff = status_file_size_in_bytes - self.__internal_file_capacity
+
+        return status_file_size_in_bytes, status_file_agent_size_diff
+
+    def __split_assessment_list(self, assessment_packages):
+        """ Split package list, keep 5 minimum packages, and remaining packages for truncation """
+        min_packages_count = Constants.StatusTruncationConfig.MIN_TRUNCATED_PACKAGE_COUNT
+        assessment_keep_five, assessment_packages_for_truncate = (assessment_packages[:min_packages_count], assessment_packages[min_packages_count:]) \
```

Comment:
rename assessment_keep_five to min_assessment_patches_to_retain and assessment_packages_for_truncate to assessment_patches_eligible_for_truncation

important note, never use a constant value in variable name, for eg, assessment_keep_five, the min assessment patch to retain count can change anytime